engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java

```diff
@@ -230,11 +229,19 @@ private void replayEvent(final LoggedEvent currentEvent) {
 
   /**
    * Ends the replay and sets some important properties, especially completes the replay future with
-   * the last source event, which has caused the last applied event.
+   * the last processing positions.
    */
   private void onRecordsReplayed() {
-    LOG.info(LOG_STMT_REPLAY_FINISHED, lastReadRecordPosition);
-    recoveryFuture.complete(lastSourceEventPosition);
+    final var lastProcessingPositions =
+        new LastProcessingPositions(
+            // the last source pos. is the last processed command by the processing state machine
+            lastSourceEventPosition,
```

Comment:
🔧  This comment describes a what not a why. It can also be captured in code by extracting the value into a local variable called `lastProcessedPosition`. It is generally considered a best practice to restrict comment usage for the why-part only.

Reply:
Yeah well I see this here as worth to mention because I know that otherwise will stumble over it and think we might put here the wrong values and even if we have a variable it is not really visible.

Reply:
Perhaps you can extract it to a variable and add a comment explaining why this can work this way